src/torchcodec/_core/FFMPEGCommon.h

```diff
@@ -50,9 +50,12 @@ struct Deleter {
 };
 
 // Unique pointers for FFMPEG structures.
-using UniqueAVFormatContext = std::unique_ptr<
+using UniqueAVFormatContextForDecoding = std::unique_ptr<
     AVFormatContext,
     Deleterp<AVFormatContext, void, avformat_close_input>>;
+using UniqueAVFormatContextForEncoding = std::unique_ptr<
+    AVFormatContext,
+    Deleter<AVFormatContext, void, avformat_free_context>>;
```

Comment:
Let's omit the preposition and call them `UniqueDecodingAVFormatContext` and `UniqueEncodingAVFormatContext`.